src/pages/index.tsx

```diff
@@ -59,6 +64,13 @@ const Home = (): JSX.Element => {
   const [shops, setShops] = useState<Array<Shop>>([]);
   const { t } = useTranslation();
 
+  useEffect(() => {
+    const initialSearchTerm = extractInitialQueryFromRouter(router);
+    if (initialSearchTerm) {
+      setQuery(initialSearchTerm);
+    }
+  }, [router]);
```

Comment:
Unfortunately due to this issue: https://github.com/zeit/next.js/issues/11201 we need a workaround, as the router is not directly ready (weirdly, one property of the router knows already of the query params...).

@jepser Do you have a better idea to prepopulate the search input field with a query parameter?

Reply:
I think it's not a big deal, I'm thinking on creating a custom hook, but it might be premature optimization, wdyt?

Reply:
Good idea with a custom hook! Maybe it’s the cleaner way, this feels a bit dirty 🙈 I actually this shouldn’t be needed, but as they don’t support this yet it seemed to be

Reply:
Done 👍